src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs

```diff
@@ -410,11 +451,11 @@ private static void HandleHistogram(TraceEvent obj, CounterConfiguration configu
             string meterName = (string)obj.PayloadValue(1);
             //string meterVersion = (string)obj.PayloadValue(2);
             string instrumentName = (string)obj.PayloadValue(3);
-            string unit = (string)obj.PayloadValue(4);
+            //string unit = (string)obj.PayloadValue(4);
```

Comment:
was this not valid for histograms?

Reply:
Duplicated essentially. For all the meters, they fire a published event. That event contains _all_ the metadata and we cache that. When a measurement happens we go look up the metadata we saw from the publish event so the unit here can be ignored. Really all the data duplication on the measurement events is kind of silly/wasteful.

PS: We also compute a DisplayName on every measurement. We could do that on publish and avoid all the allocations we're paying for on each measurement but I left it alone here.

Reply:
> Really all the data duplication on the measurement events is kind of silly/wasteful.

Its there for back-compat at this point. If the volume of events is high enough that the overhead is meaningful we could always add parallel events that were leaner, but my guess is that its neglible from the customer perspective.